### PR TITLE
Codex/ios cjk input fix

### PR DESCRIPTION
**Pull request by fufesou**
Fixes iOS soft keyboard handling for CJK IMEs in remote sessions.

The previous iOS text diff logic treated any active composing range as text that should not be sent. Some iOS IMEs keep a composing range briefly after the user confirms a candidate, so committed Chinese text could be held until another key, such as space, causes another text change.

## Changes

- Extracted iOS soft keyboard diffing into `ios_soft_keyboard_input.dart`.
- Tracks pending composing text so pinyin, stroke, bopomofo, Japanese kana, and Korean jamo/hangul composition are not sent prematurely.
- Sends committed CJK text immediately via `sessionInputString`, including Japanese kanji/kana and Korean hangul once composition is committed.
- Uses a text controller listener on iOS so commits where only the composing range changes are still observed.
- Added focused tests for Chinese, Japanese, Korean, ASCII, and backspace behavior.

## Testing

- [x] iOS soft keyboard basics: ASCII, space, return, backspace
- [x] iOS CJK IME: Korean, Chinese pinyin/stroke/bopomofo, Japanese
- [x] iOS accents and emoji smoke tests
- [x] Android soft keyboard basics: ASCII, space, return, backspace
- [x] `flutter test --no-pub test/ios_soft_keyboard_input_test.dart test/input_modifier_utils_test.dart`

## Known limitation:

- handwriting and Korean input can still expose intermediate composing text on the controlled side; this is left as-is for now because holding those events caused worse regressions

## Known issues

- Pressing multiple keys simultaneously, such as "d" and "f", will select the hidden TextFormField input, causing the Sentinel's baseline to shift. Subsequent input may result in the controlled text continuously deleting a large chunk of content, continuously inserting a large number of 1s, or deleting and then inserting.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved iOS soft-keyboard handling to avoid duplicate/missed input and better preserve composing state across Chinese, Japanese, and Korean IMEs.

* **Refactor**
  * Reworked iOS text-input flow and keyboard-visibility interactions for more reliable composition tracking and cleaner input routing.

* **Tests**
  * Added extensive unit tests validating a wide range of iOS IME composition scenarios and edge cases.

### CODE REVIEW

File: flutter/lib/mobile/utils/ios_soft_keyboard_input.dart [hunk 1]
```diff
@@ -0,0 +1,948 @@
...
+    if (!_isCompletePinyinSyllable(token.substring(0, index))) continue;
+
+    final remaining = token.substring(index);
+    if (_isPinyinSyllablePrefixWithFinal(remaining) ||
+        _isPinyinSyllableSequencePrefix(remaining)) {
+      return true;
+    }
+  }
+  return false;
+}
```
[the remark this answers is not in the document]
Comment (fufesou) [reply flutter/lib/mobile/utils/ios_soft_keyboard_input.dart:757]: I checked this path again. This only runs on short IME composing tokens, and the current logic keeps the pinyin heuristic straightforward.

Memoization would be fine, but it adds code for a theoretical case we do not have a real input trace for. I’d prefer to keep the smaller implementation here unless we see an actual performance issue.

File: flutter/test/ios_soft_keyboard_input_test.dart [hunk 2]
```diff
@@ -0,0 +1,1341 @@
...
+      previousComposingValue: previousComposingValue,
+      previousControllerText: previousControllerText,
+      sentinelPrefixLength: 3,
+    );
+    previousValue = result.nextValue;
+    previousComposingValue = result.nextComposingValue;
+    previousControllerText = currentValue;
+    return result;
+  }
+}
```
[the remark this answers is not in the document]
Comment (fufesou) [reply flutter/test/ios_soft_keyboard_input_test.dart:1239]: I’ll keep the literal `3` in these tests for now. The test data uses the explicit `111` prefix and the `TextRange(start: 3, ...)` offsets are easier to read next to that literal.

Extracting a constant would touch many cases without changing behavior, so I’d rather avoid the extra test churn.

**Comment by fufesou** [comment]
@coderabbitai review
